### ISSUE TITLE
Add support for Github, GitLab and BitBucket accounts 

### ISSUE THREAD

**Issue by craig-willis** [whole-tale/whole-tale#100]
For discussion:

* Suggest to add Github, Gitlab, and Bitbucket accounts, using their respective APIs, which would allow to interact with private repositories.
* Using SSH keys to access remote repo also doesn't seem to work. $HOME = /home/jovyan  is empty after reboot of the Tale, But SSH looks for keys in $HOME/.ssh and git cannot modify that.

While experienced users can interact with git from within a tale via command line (https://wholetale.readthedocs.io/en/stable/users_guide/github.html) including cloning private repositories using CLI token, the "create tale from Git repo" and "connect to Git repo" require public repositories.

**Comment by Xarthisius** [comment]
> * Using SSH keys to access remote repo also doesn't seem to work. $HOME = /home/jovyan  is empty after reboot of the Tale, But SSH looks for keys in $HOME/.ssh and git cannot modify that.

That's not true:
```
GIT_SSH_COMMAND="ssh -i <path_to_key>" git clone git@github.com:private_repo.git
```

**Comment by larsvilhuber** [comment]
Good to know. Not really intuitive though. Can you change that with "git config"? Would that persist across restarts of the Tale?

**Comment by Xarthisius** [comment]
> Good to know. Not really intuitive though. Can you change that with "git config"? Would that persist across restarts of the Tale?

I believe there's `core.sshCommand` in git config, but I don't recall when they introduced that. So something like this:

```
git -c core.sshCommand="ssh -i <path_to_key>" clone git@github.com:private_repo.git
```
should also work with sufficiently recent git version. Not sure if you can set it on a repo after cloning and forget about it.

For things like ssh keys, editor rcs, etc. we have `home` directory. In STATA example I believe it's mounted at `/home/jovyan/home`. It persists across all *your* Tales and it's visible only to you. We could make `git` look there by default.

**Comment by larsvilhuber** [comment]
I saw `/home/jovyan/home` but `$HOME = /home/jovyan`  which is a tad confusing for the user. SSH (and git?) will look in $HOME/(whatever), not `/home/jovyan/home` for persistent (personal) information, unless the above modifications are made each run.

**Comment by larsvilhuber** [comment]
Should work with `git config` on a per-repository level. https://docs.gitlab.com/ee/ssh/#use-different-keys-for-different-repositories

**Comment by Xarthisius** [comment]
> Should work with `git config` on a per-repository level. https://docs.gitlab.com/ee/ssh/#use-different-keys-for-different-repositories

Knowing that, would you still want UI component for uploading ssh keys, configuring repos etc. or doing via terminal is sufficient?

**Comment by larsvilhuber** [comment]
Yes, absolutely. The whole point of a GUI interface is to NOT have to do 15 different command line customizations. All of the above can be scripted from UI components, to make life easier for others.
